### PR TITLE
[mnt] slurm updates: sbatch args and requeuing

### PR DESCRIPTION
**Pull request by djarecka**
## Acknowledgment
- [x] I acknowledge that this contribution will be available under the Apache 2 license.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Summary

- tests for adding `sbatch_args` to the Submitter
- better error handling for invalid slurm options
- options for requeuing if the job is cancelled 

## Checklist

- [ ] All tests passing
- [ ] I have added tests to cover my changes
- [ ] I have updated documentation (if necessary)
- [ ] My code follows the code style of this project
(we are using `black`: you can `pip install pre-commit`,
run `pre-commit install` in the `pydra` directory
and `black` will be run automatically with each commit)

### CODE REVIEW

File: pydra/engine/workers.py [hunk 1]
```diff
@@ -296,7 +300,12 @@ async def _submit_job(self, batchscript, name, checksum, cache_dir):
             # Exception: Polling / job failure
             done = await self._poll_job(jobid)
             if done:
-                return True
+                if done == "CANCELLED":
+                    os.remove(cache_dir / f"{checksum}.lock")
```
Comment (mgxd) [inline pydra/engine/workers.py:304]:
```suggestion
                    (cache_dir / f"{checksum}.lock").unlink(missing_ok=True)
```

**Review by mgxd — COMMENTED** [review]
Hi Dorota! Does this constantly requeue if a job is cancelled, even if by the user themselves? If so, it might be worth setting this switch as a parameter when initializing a `SlurmWorker` and/or adding a limit to the number of requeues, WDYT?

**Comment by satra** [comment]
also on submitter exit, which could happen if someone hits ctrl+c we may want to actively cancel all running jobs.

@mgxd -  in most cases the auto requeue is going to be the default. but, i agree that canceling it would be good option to add to allow for users to let a job fail without disrupting the rest of the workflow. in such a case the exception associated with the job should note that the task was killed by an external process.

more generally we currently don't have a config setup in pydra, but before introducing config, i think this could be set as a worker parameter for now.

**Comment by djarecka** [comment]
@mgxd @satra - yes, right now it will go forever, so setting some max number would be good.
I was wondering also if there is a slurm command that could stop the proces or rerunning (i.e. if I could stop regardless what I set in pydra).

Also, if you have an idea how to prepare a good automatic test for it, please let me know!

**Comment by djarecka** [comment]
I was planning to add the job array in this pr, but will move it to a new one.

I hope the tests are now enough robust (this version seems to work fine on openmind and in the slurm container)

**Comment by satra** [comment]
thanks @djarecka - looks reasonable.

**Pull request merged**